### DISCUSSION TITLE
Local autotests - TypeError: unexpected keyword argument errors

### DISCUSSION THREAD

**Discussion by cnicol-gwlogic** [modflowpy/flopy#2180] [Q&A] [answered]
Can anyone provide advice on what I've done wrong with my local autotest setup? I am seeing these errors almost universally across all tests (windows 11, python 3.9, develop branch flopy as of PR #2179:
![image](https://github.com/modflowpy/flopy/assets/37314969/adce4b61-f8ab-4a48-95d2-4fb0a3309a3f)

I (think I) followed the guidance in [here](https://github.com/modflowpy/flopy/blob/develop/DEVELOPER.md) and [here](https://github.com/modflowpy/flopy/blob/develop/CONTRIBUTING.md) to no avail. It seems like a version issue to me (of something related to the autotest processes) - I've googled for it, and found similar but not exactly the same issues. I reverted pytest to 8.0.0 (from latest) to check but get the same issue.

Any tips? Thanks.

As an aside, some of the guidance needs to be updated for Ruff rather than Black etc. I'm happy to have a go at that once I sort out this local issue of mine, so I can flag for it for others.

**Comment by mwtoews** [comment]
A quick answer is that you'll need to upgrade your modflow-devtools to at least [v1.1.0](https://github.com/MODFLOW-USGS/modflow-devtools/releases/tag/1.1.0) when the strict parameter was added.

**Reply by cnicol-gwlogic** [reply]
Thanks for this @mwtoews I had quite an old version of modflow-devtools. I am however still getting a similar error:
![image](https://github.com/modflowpy/flopy/assets/37314969/57c2e29f-4c07-4f03-b560-7321614f8ffd)
![image](https://github.com/modflowpy/flopy/assets/37314969/0d99dfb7-75b5-4f44-9c6e-62013e974df9)

Any further thoughts? Cheers.

**Reply by cnicol-gwlogic** [reply]
For others: solution seems to be to downgrade pytest to pre 8.0 (i.e. 7.4.4, per [here](https://github.com/pytest-dev/pytest/issues/11868)). Thanks @mwtoews , @wpbonelli

**Reply by mwtoews** [reply, marked as answer]
You probably have `pytest-cases` installed somewhere, which interferes with pytest-8.x (#2077). Remove pytest-cases and try again.

**Reply by cnicol-gwlogic** [reply]
@mwtoews boom! You're spot on - thankyou. After removing pytest-cases, pytest 8.2.0 is working.

**Comment by wpbonelli** [comment]
**edit** echoing @mwtoews above @cnicol-gwlogic that looks like an old version of `modflow-devtools`, where the `has_pkg()` function [is coming from](https://github.com/MODFLOW-USGS/modflow-devtools/blob/748c935610c53fe90be83a6f580dc74a08f7c4df/modflow_devtools/misc.py#L362). Updating should solve it. Maybe it ought to have a mention in the developer docs..

And that's my omission re: the ruff/black transition. Thanks for raising &mdash; I can do it if you like, with some extra info on test depenencies?

**Reply by cnicol-gwlogic** [reply]
Thanks @wpbonelli go for it with the guidance updates. Thanks a lot for the quick reply too. Cheers.
